WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigActivity.kt

```diff
@@ -1,17 +1,147 @@
+        // Set max and min.
+        binding.minuteHandLengthSlider.valueTo = MINUTE_HAND_LENGTH_MAXIMUM_FOR_SLIDER
+        binding.minuteHandLengthSlider.valueFrom = MINUTE_HAND_LENGTH_MINIMUM_FOR_SLIDER
+        binding.minuteHandLengthSlider.value = MINUTE_HAND_LENGTH_DEFAULT_FOR_SLIDER
+
+        binding.minuteHandLengthSlider.addOnChangeListener { slider, value, fromUser ->
+            Log.d(TAG, "addOnChangeListener(): $slider, $value, $fromUser")
+            viewModel.setMinuteHandArmLength(value)
+        }
+
+        lifecycleScope.launch(Dispatchers.Main.immediate) {
+            viewModel.uiState.collect { uiState: WatchFaceConfigViewModel.EditWatchFaceUiState ->
+                when (uiState) {
+                    is WatchFaceConfigViewModel.EditWatchFaceUiState.Loading -> {
+                        Log.d(TAG, "StateFlow Loading: ${uiState.message}")
+                    }
+                    is WatchFaceConfigViewModel.EditWatchFaceUiState.Success -> {
+                        Log.d(TAG, "StateFlow Success.")
+                        updateWatchFacePreview(uiState.userStylesAndPreview)
+                    }
+                    is WatchFaceConfigViewModel.EditWatchFaceUiState.Error -> {
+                        Log.e(TAG, "Flow error: ${uiState.exception}")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun updateWatchFacePreview(
+        userStylesAndPreview: WatchFaceConfigViewModel.UserStylesAndPreview
+    ) {
+        Log.d(TAG, "updateWatchFacePreview: $userStylesAndPreview")
+
+        val colorStyleId: String = userStylesAndPreview.colorStyleId
+        Log.d(TAG, "\tselected color style: $colorStyleId")
+
+        binding.ticksEnabledSwitch.isChecked = userStylesAndPreview.ticksEnabled
+        binding.minuteHandLengthSlider.value = userStylesAndPreview.minuteHandLength
+        binding.preview.watchFaceBackground.setImageBitmap(userStylesAndPreview.previewImage)
+
+        if (!initialDataLoaded) {
+            initialDataLoaded = true
+            enabledWidgets()
+        }
+    }
+
+    private fun enabledWidgets() {
+        binding.colorStylePickerButton.isEnabled = true
+        binding.ticksEnabledSwitch.isEnabled = true
+        binding.minuteHandLengthSlider.isEnabled = true
+    }
+
+    override fun onDestroy() {
+        Log.d(TAG, "onDestroy()")
+        // Makes sure the activity closes.
+        finish()
+        super.onDestroy()
+    }
+
+    fun onClickColorStylePickerButton(view: View) {
+        Log.d(TAG, "onClickColorStylePickerButton() $view")
+
+        // TODO (codingjeremy): Replace with a RecyclerView to choose color style (next CL)
+        // Selects a random color style from list.
+        val colorStyleIdAndResourceIdsList = enumValues<ColorStyleIdAndResourceIds>()
+        val numbers: IntRange = colorStyleIdAndResourceIdsList.indices
+
+        val random = numbers.random()
+        val newColorStyle = colorStyleIdAndResourceIdsList[random]
```

Comment:
Small optimization here, we can just call `.random()` on the list directly I believe.